index.html

```diff
@@ -1028,6 +1061,461 @@ <h2>
+      </p>
+      <p>
+        A [=user agent=] that implements this API MUST provide a method to
+        <dfn>stop haptic effects</dfn> on an actuator. When called, if there is
+        a playing haptic effect, the [=user agent=] MUST send a command to the
+        device to stop the effect. If a haptic effect was interrupted, the
+        actuator SHOULD return to a motionless state as quickly as possible.
+      </p>
+      <section>
+        <h3>
+          Handling visibility change
+        </h3>
+        <p>
+          When the |document|'s [=Document/visibility state=] becomes `"hidden"`, run
+          these steps for each {{GamepadHapticActuator}}
+          |actuator:GamepadHapticActuator|:
+        </p>
+        <ol>
+          <li>If
+          |actuator|.{{GamepadHapticActuator/[[playingEffectPromise]]}} is
+          `undefined`, abort these steps.
+          </li>
+          <li>[=Queue a global task=] on the [=relevant global object=] of
+          |actuator| using the [=gamepad task source=] to run the following
+          steps:
+            <ol>
+              <li>[=Resolve=]
+              |actuator|.{{GamepadHapticActuator/[[playingEffectPromise]]}}
+              with {{GamepadHapticsResult/"preempted"}}.
+              </li>
+              <li>Set
+              |actuator|.{{GamepadHapticActuator/[[playingEffectPromise]]}}
+              to `undefined`.
+              </li>
+            </ol>
+          </li>
+          <li>[=Stop haptic effects=] on |actuator|.
+          </li>
+        </ol>
+      </section>
+      <section>
+        <h3>
+          Constructing a `GamepadHapticActuator`
+        </h3>
+        <p>
+          <dfn data-lt="constructing a GamepadHapticActuator">A new
+          |gamepadHapticActuator:GamepadHapticActuator|</dfn> representing a
+          {{Gamepad}}'s primary vibration actuator is constructed by
+          performing the following steps:
+        </p>
+        <ol>
+          <li>Let |gamepadHapticActuator:GamepadHapticActuator| be a newly
+          created {{GamepadHapticActuator}} instance.
+          </li>
+          <li>Let `supportedEffectsList` be an empty list.
+          </li>
+          <li>For each enum value |type:GamepadHapticEffectType| of
+          {{GamepadHapticEffectType}}, if the [=user agent=] can send a
+          command to initiate effects of that type on that actuator, append
+          |type| to `supportedEffectsList`.
+          </li>
+          <li>Set
+          |gamepadHapticActuator|.{{GamepadHapticActuator/[[effects]]}} to
+          `supportedEffectsList`.
+          </li>
+        </ol>
+      </section>
+    </section>
+    <section>
+      <h3>
+        <dfn>GamepadHapticsResult</dfn> Enum
+      </h3>
+      <p>
+        An enum representing the result of a call to
+        {{GamepadHapticActuator}}.{{GamepadHapticActuator/playEffect()}} or
+        {{GamepadHapticActuator}}.{{GamepadHapticActuator/reset()}}.
+      </p>
```

Comment:
I don't think we should say this here... the definitions automatically link back to the methods, and anywhere else these enum values are used.  

```suggestion
```